Source/WebCore/page/ElementTargetingController.cpp

```diff
@@ -648,6 +656,8 @@ void ElementTargetingController::adjustVisibilityInRepeatedlyTargetedRegions(Doc
     if (!bodyElement)
         return;
 
+    document.updateLayoutIgnorePendingStylesheets();
```

Comment:
Do we need to be concerned about this triggering script execution? Or is that no longer the case since `LocalFrameView::performPostLayoutTasks` has `ScriptDisallowedScope` now?

Reply:
I'm _pretty sure_ it's safe to update layout underneath this codepath, since other rendering update steps can trigger layout and script execution as well (and `Page::updateRendering` itself triggers layout). That said, I can omit this part from the patch for now (since it's not strictly required to implement the new methods), and add it back in a followup if needed.

I suspect there's a way to trigger a bug in this code by invalidating layout in a prior rendering update step (e.g. under `runResizeSteps`, `runScrollSteps` or `flushAutofocusCandidates`) and cause this logic to ignore potential targets.